jetty-server/src/main/java/org/eclipse/jetty/server/Response.java

```diff
@@ -621,6 +621,7 @@ public void sendError(int code, String message) throws IOException
 
         _mimeType=null;
         _characterEncoding=null;
+        _encodingFrom = EncodingFrom.NOT_SET;
```

Comment:
I think this problem was caused by this code directly setting `_characterEncoding=null;` instead of calling `setCharacterEncoding(null)`.  Probably should also call `setContentType(null);` 
Seeing a long list of reset assignments here suggests delicate code.

Reply:
There's a bunch of side effects with using `setCharacterEncoding(null)`, i did that originally, but didn't want to fall afoul of the various output mode validations that are performed within `setCharacterEncoding()`.

Perhaps we need to reset in the right order? (output mode, then character encoding, then content type) to be safe?